node_modules/col-lti/tests/test-lti.js

```diff
@@ -13,18 +13,172 @@
  * permissions and limitations under the License.
  */
 
+var _ = require('lodash');
 var assert = require('assert');
 
+var DB = require('col-core/lib/db');
 var TestsUtil = require('col-tests');
 
-describe('collabosphere', function() {
+var LtiTestsUtil = require('./util');
+var UsersTestsUtil = require('col-users/tests/util');
 
-  it('dummy test', function(callback) {
-    var client = TestsUtil.getAnonymousClient();
-    client.lti.test(function(err, response) {
-      assert.equal(response, 'COLLABOSPHERE!&');
-      return callback();
+describe('LTI', function() {
+
+  describe('Asset Library', function() {
+
+    describe('Cartridge', function() {
+
+      /**
+       * Test that verifies that the LTI cartridge can be retrieved and contains the correct information
+       */
+      it('can be retrieved', function(callback) {
+        var client = TestsUtil.getAnonymousClient();
+        LtiTestsUtil.assertAssetLibraryCartridgeSucceeds(client, function(cartridge) {
+          return callback();
+        });
+      });
     });
-  });
 
+    describe('Tool', function() {
+
+      /**
+       * Test that verifies that the Asset Library Tool can be launched
+       */
+      it('can be launched', function(callback) {
+        var client = TestsUtil.getAnonymousClient();
+        var course = TestsUtil.generateCanvasCourse(global.tests.canvas.ucberkeley);
+        var user = TestsUtil.generateCanvasUser(global.tests.canvas.ucberkeley);
+        LtiTestsUtil.assertAssetLibraryLaunchSucceeds(client, course, user, function() {
+          return callback();
+        });
+      });
+
+      /**
+       * Test that verifies that the tool validates the LTI parameters
+       */
+      it('validates the lti parameters', function(callback) {
+        var client = TestsUtil.getAnonymousClient();
+        var course = TestsUtil.generateCanvasCourse(global.tests.canvas.ucberkeley);
+        var user = TestsUtil.generateCanvasUser(global.tests.canvas.ucberkeley);
+
+        // Missing api domain
+        delete course.canvas.canvas_api_domain;
+        delete user.canvas.canvas_api_domain;
+        LtiTestsUtil.assertAssetLibraryLaunchFails(client, course, user, 400, function() {
+
+          // Missing course id
+          course = TestsUtil.generateCanvasCourse(global.tests.canvas.ucberkeley);
+          delete course.id;
+          LtiTestsUtil.assertAssetLibraryLaunchFails(client, course, user, 400, function() {
+            return callback();
+          });
+        });
+      });
+
+      /**
+       * Test that verifies that the same client can interact with multiple courses concurrently. This test
+       * will essentially ensure that the cookies that are generated by the system aren't overwriting each other
+       */
+      it('can support the same client in two courses concurrently', function(callback) {
```

Comment:
Should we have some sort of test that validates that we can successfully have launches from different api domains?